### PR TITLE
complete indexer

### PR DESCRIPTION
**Pull request by jasoncolburne**
## Rationale

There are a few things that can be cleaned up in the process of completing the `Indexer` work.

- `bexfil()` needs to be completed
- test coverage could improve
- bardages could be consolidated
- `Dater` has misc copy pasta

## Changes

- completes `bexfil()`
- improves test coverage
- consolidates bardages
- fixes a bug in `u32_to_b64()` and `u64_to_b64()`
- removes unused code from `Dater`

## Testing

```
make clean preflight
```

### CODE REVIEW

File: src/core/counter/tables.rs [hunk 1]
```diff
@@ -129,14 +118,44 @@ mod tables_tests {
...
+    #[case(&[62, 5], 2)]
+    #[case(&[62, 6], 2)]
+    #[case(&[62, 7], 2)]
+    #[case(&[62, 8], 2)]
+    #[case(&[62, 9], 2)]
+    #[case(&[62, 10], 2)]
+    #[case(&[62, 11], 2)]
+    #[case(&[62, 21], 2)]
+    #[case(&[62, 52], 3)]
+    #[case(&[62, 62], 5)]
```
Comment (jasoncolburne) [inline src/core/counter/tables.rs:154]: I thought it wise to translate the old match statement directly to tests.

File: src/core/counter/tables.rs [hunk 2]
```diff
@@ -129,14 +118,44 @@ mod tables_tests {
...
+    #[case("-C", 2)]
+    #[case("-D", 2)]
+    #[case("-E", 2)]
+    #[case("-F", 2)]
     #[case("-G", 2)]
+    #[case("-H", 2)]
+    #[case("-I", 2)]
+    #[case("-J", 2)]
+    #[case("-K", 2)]
+    #[case("-L", 2)]
```
Comment (jasoncolburne) [inline src/core/counter/tables.rs:131]: Why not.

File: src/core/util.rs [hunk 3]
```diff
@@ -177,7 +177,7 @@ pub fn u32_to_b64(n: u32, length: usize) -> Result<String> {
         out.insert(0, 'A');
     }
 
-    Ok(out)
+    Ok(out[..length].to_string())
```
Comment (jasoncolburne) [inline src/core/util.rs:180]: This was a bug! 🐛 
Caught it while playing around with `TBD0`.

File: src/core/util.rs [hunk 4]
```diff
@@ -265,6 +269,7 @@ mod util_tests {
     use rstest::rstest;
 
     #[rstest]
+    #[case(32, 0, "")]
```
Comment (jasoncolburne) [inline src/core/util.rs:272]: One of the most critical changes.

File: src/core/util.rs [hunk 5]
```diff
@@ -325,6 +330,7 @@ mod util_tests {
     #[case(&vec![252], 1, "_")]
     #[case(&vec![255, 255, 255, 255, 255, 255], 8, "________")]
     #[case(&vec![244, 0, 1], 4, "9AAB")]
+    #[case(&vec![244, 0, 1], 0, "")]
```
Comment (jasoncolburne) [inline src/core/util.rs:333]: Another critical change.

**Review by m00sey — APPROVED** [review]
lgtm  -thanks for taking the time to clean up.

We should ensure all issues are referenced so we can close them #80 for example

**Pull request merged**